File: LeetCode/Math/Golang/Leetcode829-consecutive-numbers-sum.go
package sbgo

func consecutiveNumbersSum(n int) int {
	n *= 2
	res := 0
	for b := 1; b <= n/b; b++ {
		if n%b == 0 {
			if (n/b-b+1)%2 == 0 {
				res++
			}
		}
	}
	return res
}

File: LeetCode/Math/JS/Leetcode829-consecutive-numbers-sum.js
var consecutiveNumbersSum = function (_n) {
    let [n, res] = [_n * 2, 0];
    for (let b = 1; b <= n / b; b++) {
        if (n % b == 0) {
            if ((n / b - b + 1) % 2 == 0) res++;
        }
    }
    return res;
};